app/src/main/java/com/plenart/organizeme/viewModels/SignInViewModel.kt

```diff
@@ -1,43 +1,66 @@
 package com.plenart.organizeme.viewModels
 
 import android.util.Log
+import androidx.lifecycle.LiveData
 import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
 import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.auth.FirebaseAuthException
 import com.plenart.organizeme.firebase.Firestore
 import com.plenart.organizeme.models.User
+import kotlinx.coroutines.launch
 
 class SignInViewModel: ViewModel() {
-    private lateinit var auth: FirebaseAuth;
+    private lateinit var auth: FirebaseAuth
 
-    var emailLiveData: MutableLiveData<String>? = null;
-    var passwordLiveData: MutableLiveData<String>? = null;
-    var userLiveData: MutableLiveData<User>? = null;
+    private val _email: MutableLiveData<String> = MutableLiveData()
+    private val _password: MutableLiveData<String> = MutableLiveData()
+    private val _user: MutableLiveData<User> = MutableLiveData()
 
-    init {
-        Log.i("SignInViewModel", "SignInView model created!");
-    }
-
-    fun singInUser(){
+    val email: LiveData<String>
+        get() = _email
 
-        auth = FirebaseAuth.getInstance();
+    val password: LiveData<String>
+        get() = _password
 
-        auth.signInWithEmailAndPassword(emailLiveData as String, passwordLiveData as String).addOnCompleteListener{ task ->
+    val user: LiveData<User>
+        get() = _user
 
-            if(task.isSuccessful){
-                userLiveData?.value = Firestore().loadUserDataNEW();
-                val user = auth.currentUser;
+    init {
+        Log.i("SignInViewModel", "SignInView model created!")
+    }
 
+    fun signInUser(){
+        auth = FirebaseAuth.getInstance()
+        auth.signInWithEmailAndPassword(_email.value.toString(), _password.value.toString()).addOnCompleteListener{ task ->
+            try{
+                if(task.isSuccessful){
+                    viewModelScope.launch {
+                        _user?.postValue(Firestore().loadUserData())
+                        val user = auth.currentUser
```

Comment:
I think you have no need for this. You can just have observable boolean value that is default set to false and if login is successful, set value to true.

I noticed if signIn button clicked, nothing hapended.